### PR TITLE
set external inputs and export top level connectors to resultFile

### PR DESCRIPTION
**Pull request by arun3688**
### Related Issues

#768 and also for OMEdit

### Purpose
This PR fixes bug in not exporting top level connectors to result file, and also allows users to set external inputs after initialization 

### Approach

This approach does not involve the use of `:start` suffix when using `get `or `set` methods

### CODE REVIEW

File: testsuite/OMSimulator/exportConnectorsToResultFile.lua [hunk 1]
```diff
@@ -0,0 +1,165 @@
+-- status: correct
+-- linux: yes
+-- mingw: yes
+-- win: yes
+-- mac: yes
```
Comment (lochel) [inline testsuite/OMSimulator/exportConnectorsToResultFile.lua:5]: mac: no

File: src/OMSimulatorLib/Parameters.cpp [hunk 2]
```diff
@@ -239,3 +239,49 @@ oms_status_enu_t oms::Parameters::importStartValuesHelper(pugi::xml_node& parame
...
+
+oms_status_enu_t oms::Parameters::parseModelDescription(const char *filename)
+{
+  //std::cout << "\n inside Parameters parseModelDescription ";
+  pugi::xml_document doc;
+  pugi::xml_parse_result result = doc.load_file(filename);
+  pugi::xml_node node = doc.document_element(); // modelDescription.xml
+
+  if (!result)
+    logError("Failed to load modelDescription.xml");
```
Comment (lochel) [inline src/OMSimulatorLib/Parameters.cpp:251]: return?

File: src/OMSimulatorLib/Parameters.cpp [hunk 3]
```diff
@@ -239,3 +239,49 @@ oms_status_enu_t oms::Parameters::importStartValuesHelper(pugi::xml_node& parame
...
+  {
+    std::string name = it->name();
+    //std::cout << "\n parsing xml : " << name;
+    if(name == "ModelVariables")
+    {
+      //std::cout << "\n inside ModelVariables : " << name;
+      pugi::xml_node scalarVariableNode = node.child("ModelVariables");
+      if (!scalarVariableNode)
+      {
+        logError("Error parsing modelDescription.xml");
```
Comment (lochel) [inline src/OMSimulatorLib/Parameters.cpp:264]: return?

File: src/OMSimulatorLib/Parameters.h [hunk 4]
```diff
@@ -56,10 +56,18 @@ namespace oms
...
 
     std::map<ComRef, double> realStartValues;  ///< parameters and start values defined before instantiating the FMU
     std::map<ComRef, int> integerStartValues;  ///< parameters and start values defined before instantiating the FMU
     std::map<ComRef, bool> booleanStartValues; ///< parameters and start values defined before instantiating the FMU
+
+    std::map<ComRef, double> realValues;  ///< real input values defined after initialization
+    std::map<ComRef, double> integerValues;  ///< integer input values defined after initialization
+    std::map<ComRef, double> booleanValues;  ///< boolean input values defined after initialization
+
+    std::map<ComRef, std::string> modelDescriptionStartValues; // startValues read from  modelDescription.xml
```
Comment (lochel) [inline src/OMSimulatorLib/Parameters.h:69]: use `///<`

Comment (arun3688) [reply src/OMSimulatorLib/Parameters.h]: fixed see https://github.com/OpenModelica/OMSimulator/pull/810/commits/75aa13a7b5cef1510e8b147f7f6cdd1a3d791ab5

File: src/OMSimulatorLib/Parameters.cpp (hunk 3 above)
Comment (arun3688) [reply src/OMSimulatorLib/Parameters.cpp]: fixed see https://github.com/OpenModelica/OMSimulator/pull/810/commits/75aa13a7b5cef1510e8b147f7f6cdd1a3d791ab5

File: src/OMSimulatorLib/Parameters.cpp (hunk 2 above)
Comment (arun3688) [reply src/OMSimulatorLib/Parameters.cpp]: fixed

**Review by lochel — APPROVED** [review]

**Pull request merged**